### DISCUSSION TITLE
Validation of Cognito Identity Provider Tokens ?

### DISCUSSION THREAD

**Discussion by TxMat** [awslabs/aws-sdk-rust#1048] [Q&A] [answered]
Hi,

I was wondering if there was any methods or create like [aws-jwt-verify](https://github.com/awslabs/aws-jwt-verify) to auth and verify JWT generated by cognito.

Do we have to write our own lib as a replacement ?

Thanks.

**Comment by jdisanti** [comment]
I'm not aware of anything like this existing currently in Rust.

**Comment by ivan-kiselev** [comment, marked as answer]
There are two amazing quality crates in Rust that are generic for implementing all the things auth for OAuth2 and OIDC respectively:

-    [oauth2-rs](https://github.com/ramosbugs/oauth2-rs)
-    [openidconnect-rs](https://github.com/ramosbugs/openidconnect-rs)

Both maintained by the same author, I am pretty certain they can accommodate your goals shall you look at them for long enough.
